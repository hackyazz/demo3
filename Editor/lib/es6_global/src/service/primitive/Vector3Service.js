

import * as Caml_obj from "../../../../../node_modules/bs-platform/lib/es6/caml_obj.js";
import * as Caml_int32 from "../../../../../node_modules/bs-platform/lib/es6/caml_int32.js";
import * as Matrix4Service$Wonderjs from "../../../../../node_modules/wonder.js/lib/es6_global/src/service/atom/Matrix4Service.js";
import * as Vector3Service$Wonderjs from "../../../../../node_modules/wonder.js/lib/es6_global/src/service/atom/Vector3Service.js";
import * as FloatService$WonderEditor from "../atom/FloatService.js";
import * as ValueService$WonderEditor from "../atom/ValueService.js";

function unproject(vec3, cameraToWorldMatrix, projectionMatrix) {
  return Vector3Service$Wonderjs.transformMat4Tuple(vec3, Matrix4Service$Wonderjs.multiply(cameraToWorldMatrix, Matrix4Service$Wonderjs.invert(projectionMatrix, Matrix4Service$Wonderjs.createIdentityMatrix4(/* () */0)), Matrix4Service$Wonderjs.createIdentityMatrix4(/* () */0)));
}

function multiplyScalar(param, scalar) {
  return /* tuple */[
          param[0] * scalar,
          param[1] * scalar,
          param[2] * scalar
        ];
}

function addScalar(param, scalar) {
  return /* tuple */[
          param[0] + scalar,
          param[1] + scalar,
          param[2] + scalar
        ];
}

function fromBufferAttribute(vertices, index) {
  var vIndex = Caml_int32.imul(index, 3);
  return /* tuple */[
          vertices[vIndex],
          vertices[vIndex + 1 | 0],
          vertices[vIndex + 2 | 0]
        ];
}

function distanceToSquared(param, param$1) {
  var dx = param[0] - param$1[0];
  var dy = param[1] - param$1[1];
  var dz = param[2] - param$1[2];
  return dx * dx + dy * dy + dz * dz;
}

function distanceTo(vec, v) {
  return Math.sqrt(distanceToSquared(vec, v));
}

function dot(param, param$1) {
  return param[0] * param$1[0] + param[1] * param$1[1] + param[2] * param$1[2];
}

function min(param, param$1) {
  var vz = param$1[2];
  var vy = param$1[1];
  var vx = param$1[0];
  var z = param[2];
  var y = param[1];
  var x = param[0];
  var match = Caml_obj.caml_greaterthan(x, vx);
  var match$1 = Caml_obj.caml_greaterthan(y, vy);
  var match$2 = Caml_obj.caml_greaterthan(z, vz);
  return /* tuple */[
          match ? vx : x,
          match$1 ? vy : y,
          match$2 ? vz : z
        ];
}

function max(param, param$1) {
  var vz = param$1[2];
  var vy = param$1[1];
  var vx = param$1[0];
  var z = param[2];
  var y = param[1];
  var x = param[0];
  var match = Caml_obj.caml_lessthan(x, vx);
  var match$1 = Caml_obj.caml_lessthan(y, vy);
  var match$2 = Caml_obj.caml_lessthan(z, vz);
  return /* tuple */[
          match ? vx : x,
          match$1 ? vy : y,
          match$2 ? vz : z
        ];
}

function lengthSq(param) {
  var z = param[2];
  var y = param[1];
  var x = param[0];
  return x * x + y * y + z * z;
}

function length(vec) {
  return Math.sqrt(lengthSq(vec));
}

function projectOnVector(sourceVec, targetVec) {
  var scalar = dot(targetVec, sourceVec) / lengthSq(targetVec);
  return multiplyScalar(targetVec, scalar);
}

function projectOnPlane(planeNormal, vec) {
  return Vector3Service$Wonderjs.sub(/* Float */0, vec, projectOnVector(vec, planeNormal));
}

function truncate(digit, param) {
  return /* tuple */[
          FloatService$WonderEditor.truncateFloatValue(param[0], digit),
          FloatService$WonderEditor.truncateFloatValue(param[1], digit),
          FloatService$WonderEditor.truncateFloatValue(param[2], digit)
        ];
}

function isEqual(param, param$1) {
  if (ValueService$WonderEditor.isValueEqual(/* Float */1, param$1[0], param[0]) && ValueService$WonderEditor.isValueEqual(/* Float */1, param$1[1], param[1])) {
    return ValueService$WonderEditor.isValueEqual(/* Float */1, param$1[2], param[2]);
  } else {
    return false;
  }
}

export {
  unproject ,
  multiplyScalar ,
  addScalar ,
  fromBufferAttribute ,
  distanceToSquared ,
  distanceTo ,
  dot ,
  min ,
  max ,
  lengthSq ,
  length ,
  projectOnVector ,
  projectOnPlane ,
  truncate ,
  isEqual ,
  
}
/* Matrix4Service-Wonderjs Not a pure module */
