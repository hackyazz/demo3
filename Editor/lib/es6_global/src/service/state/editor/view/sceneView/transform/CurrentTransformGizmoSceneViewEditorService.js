

import * as RecordTransformGizmoSceneViewEditorService$WonderEditor from "./RecordTransformGizmoSceneViewEditorService.js";

function mark(currentGizmoType, editorState) {
  var init = editorState[/* sceneViewRecord */3];
  var init$1 = RecordTransformGizmoSceneViewEditorService$WonderEditor.unsafeGetData(editorState);
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord : record */[
            /* viewRect */init[/* viewRect */0],
            /* gridPlane */init[/* gridPlane */1],
            /* editCamera */init[/* editCamera */2],
            /* transformGizmoData *//* record */[
              /* currentGizmoType */currentGizmoType,
              /* coordinateSystem */init$1[/* coordinateSystem */1],
              /* translationGizmoData */init$1[/* translationGizmoData */2],
              /* rotationGizmoData */init$1[/* rotationGizmoData */3],
              /* scaleGizmoData */init$1[/* scaleGizmoData */4]
            ]
          ],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function markTranslation(editorState) {
  return mark(/* Translation */0, editorState);
}

function markRotation(editorState) {
  return mark(/* Rotation */1, editorState);
}

function markScale(editorState) {
  return mark(/* Scale */2, editorState);
}

function getCurrentGizmoType(editorState) {
  return RecordTransformGizmoSceneViewEditorService$WonderEditor.unsafeGetData(editorState)[/* currentGizmoType */0];
}

export {
  mark ,
  markTranslation ,
  markRotation ,
  markScale ,
  getCurrentGizmoType ,
  
}
/* RecordTransformGizmoSceneViewEditorService-WonderEditor Not a pure module */
