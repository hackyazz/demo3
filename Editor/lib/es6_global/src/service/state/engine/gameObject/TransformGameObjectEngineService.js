

import * as TransformEngineService$WonderEditor from "../TransformEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "./GameObjectComponentEngineService.js";

function getPosition(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getPosition(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalPosition(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalPosition(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function setLocalPosition(gameObject, pos, engineState) {
  return TransformEngineService$WonderEditor.setLocalPosition(pos, GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalScale(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalScale(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function setLocalScale(gameObject, scale, engineState) {
  return TransformEngineService$WonderEditor.setLocalScale(scale, GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalToWorldMatrixTypeArray(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalToWorldMatrixTypeArray(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

export {
  getPosition ,
  getLocalPosition ,
  setLocalPosition ,
  getLocalScale ,
  setLocalScale ,
  getLocalToWorldMatrixTypeArray ,
  
}
/* TransformEngineService-WonderEditor Not a pure module */
