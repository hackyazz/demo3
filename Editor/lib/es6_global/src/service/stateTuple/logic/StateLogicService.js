

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
