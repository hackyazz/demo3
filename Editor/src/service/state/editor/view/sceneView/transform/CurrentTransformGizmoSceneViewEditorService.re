open EditorType;

open SceneViewType;

let mark = (currentGizmoType, editorState) => {
  ...editorState,
  sceneViewRecord: {
    ...editorState.sceneViewRecord,
    transformGizmoData:
      Some({
        ...
          RecordTransformGizmoSceneViewEditorService.unsafeGetData(
            editorState,
          ),
        currentGizmoType,
      }),
  },
};

let markTranslation = editorState => mark(Translation, editorState);

let markRotation = editorState => mark(Rotation, editorState);

let markScale = editorState => mark(Scale, editorState);

let getCurrentGizmoType = editorState =>
  RecordTransformGizmoSceneViewEditorService.unsafeGetData(editorState).
    currentGizmoType;