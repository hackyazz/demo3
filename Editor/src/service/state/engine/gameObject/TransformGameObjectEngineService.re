let getPosition = (gameObject, engineState) =>
  TransformEngineService.getPosition(
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );

let getLocalPosition = (gameObject, engineState) =>
  TransformEngineService.getLocalPosition(
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );

let setLocalPosition = (gameObject, pos, engineState) =>
  TransformEngineService.setLocalPosition(
    pos,
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );

let getLocalScale = (gameObject, engineState) =>
  TransformEngineService.getLocalScale(
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );

let setLocalScale = (gameObject, scale, engineState) =>
  TransformEngineService.setLocalScale(
    scale,
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );

let getLocalToWorldMatrixTypeArray = (gameObject, engineState) =>
  TransformEngineService.getLocalToWorldMatrixTypeArray(
    GameObjectComponentEngineService.unsafeGetTransformComponent(
      gameObject,
      engineState,
    ),
    engineState,
  );