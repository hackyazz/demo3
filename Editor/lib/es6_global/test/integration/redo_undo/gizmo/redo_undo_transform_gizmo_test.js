

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as EventTool$WonderEditor from "../../job/tool/EventTool.js";
import * as RedoUndoTool$WonderEditor from "../tool/RedoUndoTool.js";
import * as ReactTestTool$WonderEditor from "../../../tool/ReactTestTool.js";
import * as Vector3Service$WonderEditor from "../../../../src/service/primitive/Vector3Service.js";
import * as StateLogicService$WonderEditor from "../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as InitPickingJobTool$WonderEditor from "../../job/tool/InitPickingJobTool.js";
import * as RotationGizmosTool$WonderEditor from "../../job/tool/RotationGizmosTool.js";
import * as EventTransformGizmosTool$WonderEditor from "../../job/tool/EventTransformGizmosTool.js";
import * as BuildComponentForCurryTool$WonderEditor from "../../../tool/BuildComponentForCurryTool.js";
import * as InitTransformGizmosJobTool$WonderEditor from "../../job/tool/InitTransformGizmosJobTool.js";
import * as TransformGameObjectEngineService$WonderEditor from "../../../../src/service/state/engine/gameObject/TransformGameObjectEngineService.js";
import * as CurrentTransformGizmoSceneViewEditorService$WonderEditor from "../../../../src/service/state/editor/view/sceneView/transform/CurrentTransformGizmoSceneViewEditorService.js";

describe("redo_undo: transform gizmo", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                Curry._1(Sinon.restoreSandbox, sandbox[0]);
                return EventTool$WonderEditor.restore(/* () */0);
              }));
        describe("test translation gizmo", (function () {
                var _prepareGameObject = function (sandbox) {
                  return InitTransformGizmosJobTool$WonderEditor.prepareOneGameObject(InitPickingJobTool$WonderEditor.createCube, sandbox, 500, 200, 0, 0, /* tuple */[
                              0,
                              0,
                              3
                            ], /* tuple */[
                              0,
                              0,
                              0
                            ], /* tuple */[
                              0,
                              0,
                              0
                            ], /* () */0);
                };
                var _prepare = function (sandbox) {
                  var gameObject = _prepareGameObject(sandbox);
                  InitPickingJobTool$WonderEditor.triggerPicking(undefined, sandbox, 250, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseDown(undefined, sandbox, 260, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseMove(undefined, sandbox, 270, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseUp(undefined, undefined, undefined, sandbox, /* () */0);
                  return gameObject;
                };
                describe("test undo operate", (function () {
                        describe("test undo one step", (function () {
                                return Wonder_jest.test("\n            pick gameObject g1;\n            drag gizmo to (0.173,0,0);\n            undo;\n\n            g1->localPosition should be (0,0,0);\n            ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](Vector3Service$WonderEditor.truncate(3, StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                        return TransformGameObjectEngineService$WonderEditor.getLocalPosition(gameObject, param);
                                                                      })))), /* tuple */[
                                                          0,
                                                          0,
                                                          0
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test redo operate", (function () {
                        describe("test redo one step", (function () {
                                return Wonder_jest.test("\n            pick gameObject g1;\n            drag gizmo to (0.173,0,0);\n            undo;\n            redo\n\n            g1->localPosition should be (0.173,0,0);\n            ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              RedoUndoTool$WonderEditor.redoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](Vector3Service$WonderEditor.truncate(3, StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                        return TransformGameObjectEngineService$WonderEditor.getLocalPosition(gameObject, param);
                                                                      })))), /* tuple */[
                                                          0.173,
                                                          0,
                                                          0
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        describe("test rotation gizmo", (function () {
                var _prepare = function (sandbox) {
                  var gameObject1 = InitTransformGizmosJobTool$WonderEditor.prepareOneGameObject(InitPickingJobTool$WonderEditor.createCube, sandbox, 500, 400, 0, 0, /* tuple */[
                        0,
                        16.180339813232422,
                        11.755704879760742
                      ], /* tuple */[
                        0,
                        0,
                        0
                      ], /* tuple */[
                        0,
                        0,
                        0
                      ], /* () */0);
                  StateLogicService$WonderEditor.getAndSetEditorState(CurrentTransformGizmoSceneViewEditorService$WonderEditor.markRotation);
                  InitPickingJobTool$WonderEditor.triggerPicking(undefined, sandbox, 250, 200, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseDown(undefined, sandbox, 226, 172, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseMove(undefined, sandbox, 236, 172, /* () */0);
                  RotationGizmosTool$WonderEditor.refreshInspectorTransform(/* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseUp(undefined, undefined, undefined, sandbox, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseDown(undefined, sandbox, 236, 172, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseMove(undefined, sandbox, 246, 172, /* () */0);
                  RotationGizmosTool$WonderEditor.refreshInspectorTransform(/* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseUp(undefined, undefined, undefined, sandbox, /* () */0);
                  return gameObject1;
                };
                describe("test undo operate", (function () {
                        describe("test undo one step", (function () {
                                return Wonder_jest.test("\n               pick gameObject g1;\n               drag rotate by xy circle gizmo to r1;\n               drag rotate by xy circle gizmo to r2;\n               undo;\n\n               g1->localEulerAngle should be r1;\n               ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](InitTransformGizmosJobTool$WonderEditor.getGameObjectLocalEulerAngles(gameObject)), /* tuple */[
                                                          0,
                                                          0,
                                                          -10.4
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test redo operate", (function () {
                        describe("test redo one step", (function () {
                                return Wonder_jest.test("\n            pick gameObject g1;\n            drag rotate by xy circle gizmo to r1;\n            drag rotate by xy circle gizmo to r2;\n            undo;\n            undo;\n            redo;\n\n            g1->localEulerAngle should be r1;\n            ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              RedoUndoTool$WonderEditor.redoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](InitTransformGizmosJobTool$WonderEditor.getGameObjectLocalEulerAngles(gameObject)), /* tuple */[
                                                          0,
                                                          0,
                                                          -10.4
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("fix bug", (function () {
                        return Wonder_jest.test("\n            pick gameObject g1;\n            drag rotate by xy circle gizmo to r1;\n            drag rotate by xy circle gizmo to r2;\n            undo;\n\n            ui->inspector->transform->g1->rotation should be r1\n            ", (function () {
                                      _prepare(sandbox);
                                      RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentForCurryTool$WonderEditor.buildMainEditorTransformComponent(/* () */0));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("test scale gizmo", (function () {
                var _prepareGameObject = function (sandbox) {
                  return InitTransformGizmosJobTool$WonderEditor.prepareOneGameObject(InitPickingJobTool$WonderEditor.createCube, sandbox, 500, 200, 0, 0, /* tuple */[
                              0,
                              0,
                              3
                            ], /* tuple */[
                              0,
                              0,
                              0
                            ], /* tuple */[
                              0,
                              0,
                              0
                            ], /* () */0);
                };
                var _prepare = function (sandbox) {
                  var gameObject = _prepareGameObject(sandbox);
                  StateLogicService$WonderEditor.getAndSetEditorState(CurrentTransformGizmoSceneViewEditorService$WonderEditor.markScale);
                  InitPickingJobTool$WonderEditor.triggerPicking(undefined, sandbox, 250, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseDown(undefined, sandbox, 260, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseMove(undefined, sandbox, 270, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseUp(undefined, undefined, undefined, sandbox, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseDown(undefined, sandbox, 270, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseMove(undefined, sandbox, 290, 100, /* () */0);
                  EventTransformGizmosTool$WonderEditor.triggerMouseUp(undefined, undefined, undefined, sandbox, /* () */0);
                  return gameObject;
                };
                describe("test undo operate", (function () {
                        describe("test undo one step", (function () {
                                return Wonder_jest.test("\n            pick gameObject g1;\n            drag gizmo->x axis over +x axis to s1;\n            drag gizmo->x axis over +x axis to s2;\n            undo;\n\n            g1->localScale should be s1;\n            ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](Vector3Service$WonderEditor.truncate(3, StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                        return TransformGameObjectEngineService$WonderEditor.getLocalScale(gameObject, param);
                                                                      })))), /* tuple */[
                                                          2,
                                                          1,
                                                          1
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test redo operate", (function () {
                        describe("test redo one step", (function () {
                                return Wonder_jest.test("\n            pick gameObject g1;\n            drag gizmo->x axis over +x axis to s1;\n            drag gizmo->x axis over +x axis to s2;\n            undo;\n            undo;\n            redo;\n\n            g1->localScale should be s1;\n            ", (function () {
                                              var gameObject = _prepare(sandbox);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              RedoUndoTool$WonderEditor.redoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](Vector3Service$WonderEditor.truncate(3, StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                        return TransformGameObjectEngineService$WonderEditor.getLocalScale(gameObject, param);
                                                                      })))), /* tuple */[
                                                          2,
                                                          1,
                                                          1
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
