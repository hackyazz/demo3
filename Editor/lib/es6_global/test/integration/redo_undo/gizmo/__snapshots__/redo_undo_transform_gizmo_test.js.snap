// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: transform gizmo test rotation gizmo fix bug 
            pick gameObject g1;
            drag rotate by xy circle gizmo to r1;
            drag rotate by xy circle gizmo to r2;
            undo;

            ui->inspector->transform->g1->rotation should be r1
             1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="-10.3622"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
